src/DotNetPad/DotNetPad.Applications/Host/ScriptHost.cs

```diff
@@ -13,15 +13,16 @@ public class ScriptHost
         private byte[] loadedAssembly;
         private AppDomain scriptAppDomain;
         private RemoteScriptRun remoteScriptRun;
+        private string oldCurrentDirectory;
 
 
-        public async Task RunScriptAsync(byte[] inMemoryAssembly, byte[] inMemorySymbolStore, TextWriter outputTextWriter, TextWriter errorTextWriter, CancellationToken cancellationToken)
+        public async Task RunScriptAsync(byte[] inMemoryAssembly, byte[] inMemorySymbolStore, TextWriter outputTextWriter, TextWriter errorTextWriter, string newCurrentDirectory, CancellationToken cancellationToken)

```

Comment:
The important part of the pull request is actually this. By specifying the current directory of the script you execute to be the same directory as the script itself is saved in, you can use this tool to work on files in a similar way to python or other scripting languages.